Practical Example/IosInterviewPracticeProject/IosInterviewPracticeProject/Screens/CourseListVC.swift

```diff
@@ -0,0 +1,147 @@
+      cell.releaseDate.text = item.attributes.releasedAt.formatDateString()//.description.formatDate()
+      cell.links = item.links
+
+      if let cardArtworkUrl = URL(string: item.attributes.cardArtworkUrl) {
+        CourseAPIService.shared.fetchCardArtwork(for: cardArtworkUrl) { result in
+          switch result {
+          case .success(let artwork):
+            DispatchQueue.main.async {
+              cell.artwork.contentMode = .scaleAspectFill
+              cell.artwork.layer.cornerRadius = 5
+              cell.artwork.layer.masksToBounds = true
+              cell.artwork.image = artwork
+            }
+          case .failure(let error):
+            switch error {
+            case .imageDataError:
+              print(error.localizedDescription)
+            default:
+              print("Artwork Error: \(error.localizedDescription)")
+            }
+          }
+        }
+      }
+      return cell
+    }
+  }
+
+  private func createSnapshot(from courses: [Item]) {
+    var snapshot = NSDiffableDataSourceSnapshot<Section, Item>()
+    snapshot.appendSections([.main])
+    snapshot.appendItems(courses, toSection: .main)
+    dataSource?.apply(snapshot, animatingDifferences: true, completion: nil)
+  }
+
+  private func fetchContent() {
+    let articleUrl = "https://raw.githubusercontent.com/raywenderlich/ios-interview/master/Practical%20Example/articles.json"
+    let videoUrl = "https://raw.githubusercontent.com/raywenderlich/ios-interview/master/Practical%20Example/videos.json"
+    let fetchContentDispatchGroup = DispatchGroup()
+
+    fetchContentDispatchGroup.enter()
+    CourseAPIService.shared.fetchContent(for: articleUrl) { results in
+      switch results {
+      case .success(let results):
+        self.courses.append(contentsOf: results.data)
+        fetchContentDispatchGroup.leave()
+      case .failure(let error):
+        print("Failed: \(error.localizedDescription)")
+      }
+    }
+
+    fetchContentDispatchGroup.enter()
+    CourseAPIService.shared.fetchContent(for: videoUrl) { results in
+      switch results {
+      case .success(let results):
+        self.courses.append(contentsOf: results.data)
+        fetchContentDispatchGroup.leave()
+      case .failure(let error):
+        print("Failed: \(error.localizedDescription)")
+      }
+    }
+
+    fetchContentDispatchGroup.notify(queue: .main) {
+      let coursesSortedByDateDescending = self.courses.sorted { $0.attributes.releasedAt > $1.attributes.releasedAt }
+      self.createSnapshot(from: coursesSortedByDateDescending)
+    }
+  }
+}
+
+extension CourseListVC: UITableViewDelegate {
+  override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+    guard let courseListCell = sender as? CourseListCell else { return }
+
+    if let courseContentUrl = courseListCell.links?.current {
+      if segue.identifier == StoryboardSegue.CourseDetailsSegue {
+        if let courseDetailsVC = segue.destination as? CourseDetailsVC {
```

Comment:
You can comma-separate these ifs instead of adding them one inside the other.

Reply:
fixed